### ISSUE TITLE
React Native iOS build failed because "'simdjson.h' file not found"

### ISSUE THREAD

**Issue by backmeupplz** [Nozbe/WatermelonDB#1078]
I'm not sure what's going on. My RN version is `0.64.2`, watermelondb version is `0.23.0-9`

![Screen Shot 2021-06-24 at 8 52 06 AM](https://user-images.githubusercontent.com/3192028/123294184-774d4600-d4c9-11eb-9c93-8d33dcce22b1.png)

**Comment by backmeupplz** [comment]
Same happens on watermelondb version `0.23.0-11`

**Comment by backmeupplz** [comment]
Here's my Podfile in case you need it:

```
platform :ios, '12.0'
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

inhibit_all_warnings!

target 'Todorant' do
  # React native
  config = use_native_modules!
  use_react_native!(:path => config["reactNativePath"])

  # The rest
  use_native_modules!
  pod 'RNReanimated', :path => '../node_modules/react-native-reanimated'

  pod 'RNGestureHandler', :path => '../node_modules/react-native-gesture-handler'
  pod 'react-native-camera', :path => '../node_modules/react-native-camera'
  
  permissions_path = '../node_modules/react-native-permissions/ios'
  pod 'Permission-Camera', :path => "#{permissions_path}/Camera.podspec"
  pod 'Permission-Siri', :path => "#{permissions_path}/Siri.podspec"
  
  pod 'RNVectorIcons', :path => '../node_modules/react-native-vector-icons'

  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi', :modular_headers => true
end

target 'TodorantToday' do
  pod 'Alamofire', '~> 5.0'
  pod 'SwiftKeychainWrapper'
end

target 'TodorantHomescreen' do
  pod 'Alamofire', '~> 5.0'
  pod 'SwiftKeychainWrapper'
end

target 'TodorantIntents' do
  pod 'Alamofire', '~> 5.0'
  pod 'SwiftKeychainWrapper'
end

target 'Todorant-watchOS Extension' do
  platform :watchos, '7.0'
  pod 'Alamofire', '~> 5.0'
  pod 'KeychainAccess'
end

post_install do |pi|
  pi.pods_project.targets.each do |t|
    t.build_configurations.each do |config|
      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'
    end
  end
end
```

An my bridging header:

```
#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>
#import <React/RCTViewManager.h>
#import <React/RCTBridgeModule.h>

#import "../../node_modules/@nozbe/watermelondb/native/ios/WatermelonDB/SupportingFiles/Bridging.h"
```

**Comment by backmeupplz** [comment]
I've got the dependency in the `yarn.lock`, but not in the podfile lock 🤔

![Screen Shot 2021-06-24 at 9 02 05 AM](https://user-images.githubusercontent.com/3192028/123295734-dc556b80-d4ca-11eb-8981-d37dce7b65ef.png)

**Comment by backmeupplz** [comment]
Hm, it doesn't look like `simdjson.h` is a part of the binary

**Comment by backmeupplz** [comment]
Version `0.22.0` works well though.

**Comment by radex** [comment]
@backmeupplz Please read the changelog-unreleased & installation guide more carefully, and then tell me if you still run into troubles

**Comment by kevinwcyu** [comment]
@rodrigogaraujo, have you run `pod install` in your `ios` directory?

**Comment by ebnersilva** [comment]
Hi,

My Environment:

React native: 0.65.1
Watermelon: 0.23.0

I was having the same problem. For me the problem was gone when i did two steps:  

First: in the `project_name-Bridging-Header.h` i changed this part:

`#import "../../node_modules/@nozbe/watermelondb/native/ios/WatermelonDB/SupportingFiles/Bridging.h"`

for this

`#import "../node_modules/@nozbe/watermelondb/native/ios/WatermelonDB/SupportingFiles/Bridging.h"`

Second: i open `projectname.xcworkspace` with xcode and run my project with success. 

Before I was opening the project file `projectname.xcodeproj` and i  did receiving this error.

or you can compile the project with success running the command: 

`yarn ios or npm run ios` in the root folder project

**Comment by aleksandraszpila** [comment]
I followed steps from installation guide and I've got the same problem using watermelondb version 0.24.0, any solution for that?

**Comment by tom-krusch** [comment]
+1

**Comment by AndreCosta101** [comment]
Same problem here.

**Comment by jamesjlv** [comment]
Hello guys,

This is what's worked for me:


1. Go to node_modules/@nozbe/simdjson/src

      - Copy those 2 files “simdjson.cpp” and “simdjson.h”
      
![Captura de Tela 2021-12-02 às 00.41.08.png](https://xesque.rocketseat.dev/forum/1638416485749-Captura-de-Tela-2021-12-02-às-00.41.08.png)


2. Go to node_modules/@nozbe/watermelondb/native/shared

      - Paste the files that you copied above  “simdjson.cpp” and “simdjson.h”
      
![Captura de Tela 2021-12-02 às 00.39.50.png](https://xesque.rocketseat.dev/forum/1638416440478-Captura-de-Tela-2021-12-02-às-00.39.50.png)

3. Execute the comand yarn run ios , for me worked.


### Make sure that you edited your podfile and write the 2 lines below:

```
    pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi', :modular_headers => true
    pod 'simdjson', path: '../node_modules/@nozbe/simdjson'

```
_Font: https://nozbe.github.io/WatermelonDB/Installation.html_

### Some erros shows up, and this is how I fix:

* In the first time, I got a message (fatal error: malformed or corrupted AST file…), with this, I have to go in the /Users/YOURUSERNAME/Library/Developer/Xcode/DerivedData and clean all the files in this folder before a try to make a new compilation.

* To make sure that everything will works well, I deleted my folder Pods inside tha "ios" folder, and after I execute the command pod install to build this folder again. (*Make sure you have the backup*)


* I deleted my folder node_modules to and download again with the command yarn install.


I hope this helps someone, bye bye ;)

**Comment by meedwire** [comment]
+1

**Comment by programminPete** [comment]
> Here's my Podfile in case you need it:
> 
> ```
> platform :ios, '12.0'
> require_relative '../node_modules/react-native/scripts/react_native_pods'
> require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
> 
> inhibit_all_warnings!
> 
> target 'Todorant' do
>   # React native
>   config = use_native_modules!
>   use_react_native!(:path => config["reactNativePath"])
> 
>   # The rest
>   use_native_modules!
>   pod 'RNReanimated', :path => '../node_modules/react-native-reanimated'
> 
>   pod 'RNGestureHandler', :path => '../node_modules/react-native-gesture-handler'
>   pod 'react-native-camera', :path => '../node_modules/react-native-camera'
>   
>   permissions_path = '../node_modules/react-native-permissions/ios'
>   pod 'Permission-Camera', :path => "#{permissions_path}/Camera.podspec"
>   pod 'Permission-Siri', :path => "#{permissions_path}/Siri.podspec"
>   
>   pod 'RNVectorIcons', :path => '../node_modules/react-native-vector-icons'
> 
>   pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi', :modular_headers => true
> end
> 
> target 'TodorantToday' do
>   pod 'Alamofire', '~> 5.0'
>   pod 'SwiftKeychainWrapper'
> end
> 
> target 'TodorantHomescreen' do
>   pod 'Alamofire', '~> 5.0'
>   pod 'SwiftKeychainWrapper'
> end
> 
> target 'TodorantIntents' do
>   pod 'Alamofire', '~> 5.0'
>   pod 'SwiftKeychainWrapper'
> end
> 
> target 'Todorant-watchOS Extension' do
>   platform :watchos, '7.0'
>   pod 'Alamofire', '~> 5.0'
>   pod 'KeychainAccess'
> end
> 
> post_install do |pi|
>   pi.pods_project.targets.each do |t|
>     t.build_configurations.each do |config|
>       config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'
>     end
>   end
> end
> ```
> 
> And my bridging header:
> 
> ```
> #import <React/RCTBundleURLProvider.h>
> #import <React/RCTRootView.h>
> #import <React/RCTViewManager.h>
> #import <React/RCTBridgeModule.h>
> 
> #import "../../node_modules/@nozbe/watermelondb/native/ios/WatermelonDB/SupportingFiles/Bridging.h"
> ```

You are missing the other two required pod lines - from the installation guide, section 3. : https://nozbe.github.io/WatermelonDB/Installation.html


`# If you're using autolinking, this line might not be needed
pod 'WatermelonDB', :path => '../node_modules/@nozbe/watermelondb'

# NOTE: Do not remove, needed to keep WatermelonDB compiling:
pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi', :modular_headers => true

# NOTE: This is required as of v0.23
pod 'simdjson', path: '../node_modules/@nozbe/simdjson'
`

hope that helps.
Also. I had the same error but was because I had the pod install outside of the correct target . Once I put all 3 of those inside the main target, yarn install / cd ios && pod install  worked as expected.

**Comment by renatobentorocha** [comment]
> Hello guys,
> 
> This is what's worked for me:
> 
> 1. Go to node_modules/@nozbe/simdjson/src
>    
>    * Copy those 2 files “simdjson.cpp” and “simdjson.h”
> 
> ![Captura de Tela 2021-12-02 às 00.41.08.png](https://camo.githubusercontent.com/0472cf59be89411748fa29d3b8502ffd64dcc79d4901d088b8d52a07f862b063/68747470733a2f2f7865737175652e726f636b6574736561742e6465762f666f72756d2f313633383431363438353734392d436170747572612d64652d54656c612d323032312d31322d30322d61254343253830732d30302e34312e30382e706e67)
> 
> 2. Go to node_modules/@nozbe/watermelondb/native/shared
>    
>    * Paste the files that you copied above  “simdjson.cpp” and “simdjson.h”
> 
> ![Captura de Tela 2021-12-02 às 00.39.50.png](https://camo.githubusercontent.com/e6cd57f7f42ae4a87754d7cfce7d21094ec65bda5f048449241ba7ed56cdf519/68747470733a2f2f7865737175652e726f636b6574736561742e6465762f666f72756d2f313633383431363434303437382d436170747572612d64652d54656c612d323032312d31322d30322d61254343253830732d30302e33392e35302e706e67)
> 
> 3. Execute the comand yarn run ios , for me worked.
> 
> ### Make sure that you edited your podfile and write the 2 lines below:
> ```
>     pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi', :modular_headers => true
>     pod 'simdjson', path: '../node_modules/@nozbe/simdjson'
> ```
> 
> _Font: https://nozbe.github.io/WatermelonDB/Installation.html_
> 
> ### Some erros shows up, and this is how I fix:
> * In the first time, I got a message (fatal error: malformed or corrupted AST file…), with this, I have to go in the /Users/YOURUSERNAME/Library/Developer/Xcode/DerivedData and clean all the files in this folder before a try to make a new compilation.
> * To make sure that everything will works well, I deleted my folder Pods inside tha "ios" folder, and after I execute the command pod install to build this folder again. (_Make sure you have the backup_)
> * I deleted my folder node_modules to and download again with the command yarn install.
> 
> I hope this helps someone, bye bye ;)

Hi @jamesjlv thanks a lot by the solution. I have added a little command at xcode that did the same that you did.

![Captura de Tela 2022-10-19 às 10 45 59](https://user-images.githubusercontent.com/18239349/197058901-e2b4a751-9cfa-4b0f-a930-fb36844e1334.png)

**Comment by chiubaka** [comment]
For those looking to copy pasta the script in the above screenshot:
```
cp "${PROJECT_DIR}/../node_modules/@nozbe/simdjson/src/simdjson.cpp" "${PROJECT_DIR}/../node_modules/@nozbe/watermelondb/native/shared/simdjson.cpp"

cp "${PROJECT_DIR}/../node_modules/@nozbe/simdjson/src/simdjson.h" "${PROJECT_DIR}/../node_modules/@nozbe/watermelondb/native/shared/simdjson.h"
```

**Comment by radex** [comment]
Should be fixed in 0.26

**Comment by faerylay** [comment]
# WatermelonDB requires simdjson from @nozbe/simdjson (not CocoaPods registry)
  pod 'simdjson', path: '../node_modules/@nozbe/simdjson'
